### PR TITLE
Cleanup docstrings for `gammapy.astro`

### PR DESCRIPTION
**Pull request by Astro-Kirsty**
This addresses part of issue #4811, specifically for `gammapy.astro `

### CODE REVIEW

File: gammapy/astro/population/spatial.py [hunk 1]
```diff
@@ -316,19 +316,20 @@ def xy_position(self, theta=None, radius=None, spiralarm_index=0):
...
+            Spiral arm index. Default is 0.
 
         Returns
         -------
-        x, y : array_like
+        x, y : `~numpy.ndarray`
             Position (x, y)
         """
         if (theta is None) and not (radius is None):
+            print(self.theta)
```
Comment (QRemy) [inline gammapy/astro/population/spatial.py:332]:
```suggestion
```

File: gammapy/astro/population/simulate.py [hunk 2]
```diff
@@ -151,17 +151,17 @@ def make_base_catalog_galactic(
...
+        Radial surface density distribution of sources.
     vel_dis : callable
-        Proper motion velocity distribution of sources
+        Proper motion velocity distribution of sources.
     max_age : `~astropy.units.Quantity`
-        Maximal age of the source
+        Maximal age of the source.
     spiralarms : bool
-        Include a spiralarm model in the catalog?
+        Include a spiralarm model in the catalog.
```
Comment (QRemy) [inline gammapy/astro/population/simulate.py:162]:
```suggestion
        Include a spiral arm model in the catalog.
```

**Review by QRemy — DISMISSED** [review]
Thanks a lot @Astro-Kirsty !

File: gammapy/astro/population/spatial.py (hunk 1 above)
Comment (Astro-Kirsty) [reply gammapy/astro/population/spatial.py:332]: Oops thanks!

**Review by cgalelli — DISMISSED** [review]
Thanks for the work @Astro-Kirsty !

File: gammapy/astro/population/simulate.py [hunk 3]
```diff
@@ -384,14 +384,15 @@ def add_observed_parameters(table, obs_pos=None):
     Parameters
     ----------
     table : `~astropy.table.Table`
-        Input table
+        Input table.
     obs_pos : tuple or None
```
Comment (MRegeard) [inline gammapy/astro/population/simulate.py:388]: Here, should we put `tuple, optional` and then in the description explain that if `None`, default to Earth position ?

**Review by MRegeard — DISMISSED** [review]
Thanks @Astro-Kirsty !
See inline comment.

File: gammapy/astro/population/simulate.py (hunk 3 above)
Comment (Astro-Kirsty) [reply gammapy/astro/population/simulate.py:388]: Thanks @MRegeard 
That's a nice idea

File: gammapy/astro/darkmatter/spectra.py [hunk 4]
```diff
@@ -153,6 +153,7 @@ def channel(self, channel):
             self._channel = channel
 
     def evaluate(self, energy, **kwargs):
+        """Evaluate the PrimaryFlux."""
```
Comment (MRegeard) [inline gammapy/astro/darkmatter/spectra.py:156]: Shouldn't `PrimaryFlux` be given as a reference ? i.e. :

```suggestion
        """Evaluate the `~gammapy.astro.darkmatter.PrimaryFlux`."""
```

**Review by MRegeard — DISMISSED** [review]
Thanks @Astro-Kirsty, seems good to me. Just take a look at my inline suggestion.

File: gammapy/astro/darkmatter/spectra.py (hunk 4 above)
Comment (Astro-Kirsty) [reply gammapy/astro/darkmatter/spectra.py:156]: Keeping in line with the other evaluate functions I don't think we should reference the class inside the class. But I have adapted it slightly

File: gammapy/astro/population/spatial.py [hunk 5]
```diff
@@ -189,8 +188,8 @@ def evaluate(r, amplitude, a, b):
 
 
 class FaucherKaspi2006(Fittable1DModel):
-    r"""Radial distribution of the birth surface density of pulsars in the galaxy
-     - Faucher-Giguere & Kaspi 2006.
+    r"""
+    Radial distribution of the birth surface density of pulsars in the galaxy - Faucher-Giguere & Kaspi 2006.
```
Comment (cgalelli) [inline gammapy/astro/population/spatial.py:192]: Should we start using a `References` section at the end of docstrings?

File: gammapy/astro/population/spatial.py [hunk 6]
```diff
@@ -316,16 +315,16 @@ def xy_position(self, theta=None, radius=None, spiralarm_index=0):
...
-        theta : array_like
-            Angle (deg)
-        radius : array_like
-            Radius (kpc)
-        spiralarm_index : int
-            Spiral arm index
+        theta : `~astropy.units.Quantity`, optional
+            Angle (deg). Default is None.
+        radius : `~astropy.units.Quantity`, optional
+            Radius (kpc). Default is None
```
Comment (cgalelli) [inline gammapy/astro/population/spatial.py:321]:
```suggestion
            Radius (kpc). Default is None.
```

File: gammapy/astro/population/spatial.py [hunk 7]
```diff
@@ -367,14 +366,14 @@ def theta(self, radius, spiralarm_index):
...
+            Radius (kpc).
         spiralarm_index : int
-            Spiral arm index
+            Spiral arm index.
 
         Returns
         -------
-        theta : array_like
+        theta : `~numpy.ndarray`
             Angle (deg)
```
Comment (cgalelli) [inline gammapy/astro/population/spatial.py:377]:
```suggestion
            Angle (deg).
```

File: gammapy/astro/population/spatial.py [hunk 8]
```diff
@@ -408,14 +407,16 @@ def _blur(radius, theta, amount=0.07, random_state="random-seed"):
         Parameters
         ----------
         radius : `~astropy.units.Quantity`
-            Radius coordinate
+            Radius coordinate.
         theta : `~astropy.units.Quantity`
-            Angle coordinate
+            Angle coordinate.
         amount: float, optional
```
Comment (cgalelli) [inline gammapy/astro/population/spatial.py:413]:
```suggestion
        amount : float, optional
```

File: gammapy/astro/population/spatial.py (hunk 5 above)
Comment (Astro-Kirsty) [reply gammapy/astro/population/spatial.py:192]: Could be a good idea!

File: gammapy/astro/darkmatter/spectra.py (hunk 4 above)
Comment (MRegeard) [reply gammapy/astro/darkmatter/spectra.py]: Yes, I agree then !

**Review by registerrier — APPROVED** [review]
Thanks @Astro-Kirsty . Looks good.

**Comment by registerrier** [comment]
Let's try to backport.

**Pull request merged**